### PR TITLE
Fix host network mode detection

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Fix host network mode detection

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by shang-wang — APPROVED** [review]
Yay

File: pkg/util/docker/host_ip_provider.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ type hostIPProvider struct {
 
 func getDockerHostIPsUncached() []string {
 	var isHostMode bool
-	if mode, err := GetAgentContainerNetworkMode(); err != nil && mode == "host" {
+	if mode, err := GetAgentContainerNetworkMode(); err == nil && mode == "host" {
```
Comment (ahmed-mez) [inline pkg/util/docker/host_ip_provider.go]:
```suggestion
	if mode, err := GetAgentContainerNetworkMode(); err == nil && mode == containers.HostNetworkMode {
```

**Review by ahmed-mez — APPROVED** [review]

File: pkg/util/docker/host_ip_provider.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ type hostIPProvider struct {
 
 func getDockerHostIPsUncached() []string {
 	var isHostMode bool
-	if mode, err := GetAgentContainerNetworkMode(); err != nil && mode == "host" {
+	if mode, err := GetAgentContainerNetworkMode(); err == nil && mode == containers.HostNetworkMode {
```
Comment (kevinconaway) [inline pkg/util/docker/host_ip_provider.go]: Are we missing test coverage here ?

Comment (p-lambert) [reply pkg/util/docker/host_ip_provider.go]: we are. unfortunately is is tricky to test this as it depends on runtime environment.

**Pull request merged**